features/mbedtls/platform/inc/platform_mbed.h

```diff
@@ -20,3 +20,11 @@
 #if defined(DEVICE_TRNG)
 #define MBEDTLS_ENTROPY_HARDWARE_ALT
 #endif
+
+/*
+ * Enable mbed TLS to use mbed OS mutexes
+ */
+#if defined(MBEDTLS_THREADING)
```

Comment:
With `MBEDTLS_THREADING` we're effectively defining a new mbed OS macro to control mbed TLS. I would like to put it in a separate namespace from the normal mbed TLS configuration macros to make it more obvious.

Could we use something like `TLS_CONFIG_XXXX`? I'm open to ideas.